lib/src/jwt.dart

```diff
@@ -281,7 +307,8 @@ class JWT {
     bool noIssueAt = false,
   }) {
     try {
-      if (payload is Map<String, dynamic> || payload is Map<dynamic, dynamic>) {
+      var payload = this.payload;
+      if (payload is Map<String, dynamic>) {
         try {
           payload = Map<String, dynamic>.from(payload);
```

Comment:
I think it made sense that this was taking a copy of the map (as it might've been immutable or used again in the previous variant which was not holding it's own copy), but I wonder if the "writing back" to `payload` was intentional?

Now we don't write back to the JWT object (`this`) (and the local reference is also needed to get the type checking work properly, as it's now based on Object instead of the "always working" `dynamic`).